### PR TITLE
FEAT: Improved auto_parametrize_design method.

### PR DESCRIPTION
**Pull request by maxcapodi78**
Variable renaming and also the trace_net_filters now applies also to padstack definitions.

### CODE REVIEW

File: src/pyedb/dotnet/edb.py [hunk 1]
```diff
@@ -4122,20 +4123,25 @@ def auto_parametrize_design(
...
         List(str)
             List of all parameters name created.
         """
+        if isinstance(trace_net_filter, str):
+            trace_net_filter = [trace_net_filter]
         parameters = []
         if layers:
             if not layer_filter:
-                _layers = self.stackup.stackup_layers
+                _layers = self.stackup.layers
```
Comment (svandenb-dev) [inline src/pyedb/dotnet/edb.py:4140]: Why are we taking also non stackup layers ? I am nit sure we can apply parameters on them

File: src/pyedb/dotnet/edb.py [hunk 2]
```diff
@@ -4122,20 +4123,25 @@ def auto_parametrize_design(
...
             else:
                 if isinstance(layer_filter, str):
                     layer_filter = [layer_filter]
-                _layers = {k: v for k, v in self.stackup.stackup_layers.items() if k in layer_filter}
+                _layers = {k: v for k, v in self.stackup.layers.items() if k in layer_filter}
```
Comment (svandenb-dev) [inline src/pyedb/dotnet/edb.py:4144]: same question

**Review by svandenb-dev — APPROVED** [review]
LGTM

File: src/pyedb/dotnet/edb.py (hunk 1 above)
Comment (maxcapodi78) [reply src/pyedb/dotnet/edb.py:4172]: stackup_layers is deprecated
it returns layers

File: src/pyedb/dotnet/edb.py (hunk 2 above)
Comment (maxcapodi78) [reply src/pyedb/dotnet/edb.py:4176]: image

**Review by svandenb-dev — APPROVED** [review]
LGTM

**Review by svandenb-dev — APPROVED** [review]
LGTM

**Pull request merged**